src/braintree-manager/payment-providers/apple-pay/apple-pay.ts

```diff
@@ -19,31 +19,36 @@ export class ApplePayHandler implements ApplePayHandlerInterface {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   instance: PromisedSingleton<any | undefined>;
 
-  constructor(
-    braintreeManager: BraintreeManagerInterface,
-    applePayClient: braintree.ApplePay,
-    applePaySessionManager: ApplePaySessionManagerInterface,
-  ) {
-    this.braintreeManager = braintreeManager;
-    this.applePayClient = applePayClient;
-    this.applePaySessionManager = applePaySessionManager;
+  constructor(options: {
+    braintreeManager: BraintreeManagerInterface;
+    applePayClient: braintree.ApplePay;
+    applePaySessionManager: ApplePaySessionManagerInterface;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    instancePromisedSingleton?: PromisedSingleton<any | undefined>;
+  }) {
+    this.braintreeManager = options.braintreeManager;
+    this.applePayClient = options.applePayClient;
+    this.applePaySessionManager = options.applePaySessionManager;
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    this.instance = new PromisedSingleton<any | undefined>({
-      generator: new Promise(resolve => {
-        if (!this.applePaySessionManager.canMakePayments()) {
-          return resolve(undefined);
-        }
-
-        return this.braintreeManager.instance.get().then(async braintreeClient => {
-          const instance = await this.applePayClient.create({
-            client: braintreeClient,
+    this.instance =
+      options.instancePromisedSingleton ??
```

Comment:
ah, so we are only loading if available?

Reply:
That is true, but what this line is doing is letting us pass in an override for the singleton so we can mock it out during unit testing. Like [here](https://github.com/internetarchive/iaux-donation-form/pull/1/files/d0e3d6f6988190f2728822077f2015616f58b815..274ba3ca97520e595560f88d061d3fde49ac7a4f#diff-aff7be17656c7589b24c75d2a4625e9cR49-R59) I'm testing if ApplePay is unavailable so I pass in a promise that rejects so ApplePay is unavailable.